SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp

```diff
@@ -0,0 +1,272 @@
+  template <typename DataT, int NumElems>
+  static void
+  call_simd_by_const_ref(const esimd::simd<DataT, NumElems> &simd_by_const_ref,
+                         DataT *const out) {
+    simd_by_const_ref.copy_to(out);
+  }
+};
+
+// Test case description
+template <int VecSize, typename ContextT>
+struct LoadCtorAccTestDescription : public ITestDescription {
+  using BaseT = ctors::LoadCtorTestDescription<VecSize, ContextT>;
+
+public:
+  LoadCtorAccTestDescription(const std::string &vec_data_name, int dims,
+                             const std::string &acc_mode_name,
+                             const std::string &acc_target_name,
+                             unsigned int offset,
+                             const std::string &alignment_name) {
+    std::ostringstream stream;
+
+    using BaseDescriptionT = ctors::LoadCtorTestDescription<VecSize, ContextT>;
+    const BaseDescriptionT base_description(vec_data_name, alignment_name);
+    const AccessorDescription accessor_description(
+        vec_data_name, dims, acc_mode_name, acc_target_name);
+
+    stream << base_description.to_string();
+    stream << ", with offset " << log::stringify(offset);
+    stream << ", from " << accessor_description.to_string();
+
+    m_description = stream.str();
+  }
+
+  std::string to_string() const override { return m_description; }
+
+private:
+  std::string m_description;
+};
+
+// The main test routine
+//
+template <typename DataT, typename VecSizeT, typename AccDimsT,
+          typename AccModeT, typename AccTargetT, typename ContextT,
+          typename OffsetGeneratorT, typename AlignmentT>
+class run_test {
+  static constexpr int VecSize = VecSizeT::value;
+  static constexpr int AccDims = AccDimsT::value;
+  static constexpr sycl::access_mode AccMode = AccModeT::value;
+  static constexpr sycl::target AccTarget = AccTargetT::value;
+  static constexpr auto Offset =
+      OffsetGeneratorT::template get<AlignmentT, DataT, VecSize>();
+
+  using SimdT = esimd::simd<DataT, VecSize>;
+  using AccessorT = sycl::accessor<DataT, AccDims, AccMode, AccTarget>;
+  using TestDescriptionT = LoadCtorAccTestDescription<VecSize, ContextT>;
+  using KernelT = Kernel<DataT, VecSize, AccDimsT, AccModeT, AccTargetT,
+                         ContextT, OffsetGeneratorT, AlignmentT>;
+
+  static_assert(AccTarget == sycl::target::device,
+                "Accessor target is not supported");
+
+public:
+  template <typename... TestDescriptionArgsT>
+  bool operator()(sycl::queue &queue, const std::string &vec_data_name,
+                  const std::string &acc_mode_name,
+                  const std::string &acc_target_name,
+                  const std::string &alignment_name) {
+    // Define the mapping between parameters retrieved and test descriptor
+    // arguments
+    return run(queue, /* The rest are the test description parameters */
+               vec_data_name, AccDims, acc_mode_name, acc_target_name, Offset,
+               alignment_name);
```

Comment:
Do we need the `run` function?

Reply:
The intent here is to have a single place where the mapping between `operator()` parameters and `LoadCtorAccTestDescription` parameters is done. Anything after `/* The rest are the test description parameters */` is passed as-is into the test  case description within the `run()`, so we don't need to manually handle parameter order each time we construct the test case description:
```
log::trace<TestDescriptionT>(std::forward<TestDescriptionArgsT>(args)...);
```

BTW, looks like I've left template parameters for `operator()` unintentionally. It's definitely confusing, sorry for that - will fix ASAP.

Reply:
Removed template parameters for the `operator()` in 46f0ea82b55123bb5abd90243d280057bea8c51f

Reply:
ok